src/charmed_kubeflow_chisme/testing/cos_integration.py

```diff
@@ -133,23 +135,39 @@ async def _get_relation(app: Application, endpoint_name: str) -> Relation:
     return relations[0]
 
 
-async def _get_app_relation_data(app: Application, endpoint_name: str) -> Dict[str, Any]:
+def _get_app_from_relation(relation: Relation, side: str) -> Application:
+    """Get application from relation."""
+    if side == PROVIDES:
+        return relation.provides.application
+    elif side == REQUIRES:
+        return relation.requires.application
+
+    raise ValueError(f"{side} is invalid side of relation.")
+
+
+async def _get_app_relation_data(
+    app: Application, endpoint_name: str, side: str
+) -> Dict[str, Any]:
     """Get application relation data from endpoint name."""
     relation = await _get_relation(app, endpoint_name)
-    unit = app.units[0]  # Note(rgildein) use first unit, since we are getting application data
-    cmd = f"relation-get --format=yaml -r {relation.entity_id} --app - {app.name}"
+    relation_app = _get_app_from_relation(relation, side)
```

Comment:
Should we add a comment about this change? Why we need to use get_app_from_relation instead of just app.units[0]?